core/platform/email/mailgun_email_services.py

```diff
@@ -118,28 +124,35 @@ def send_email_to_recipients(
         # email to each recipient (This is intended to be a workaround for
         # sending individual emails).
         data['recipient_variables'] = recipient_variables or {}
-
-        # The b64encode accepts and returns bytes, so we first need to encode
-        # the MAILGUN_API_KEY to bytes, then decode the returned bytes back
-        # to string.
-        base64_mailgun_api_key = base64.b64encode(
-            b'api:%b' % mailgun_api_key.encode('utf-8')
-        ).strip().decode('utf-8')
-        auth_str = 'Basic %s' % base64_mailgun_api_key
-        header = {'Authorization': auth_str}
         server = 'https://api.mailgun.net/v3/%s/messages' % (
             feconf.MAILGUN_DOMAIN_NAME
         )
-        # The 'ascii' is used here, because only ASCII char are allowed in url,
-        # also the docs recommend this approach:
-        # https://docs.python.org/3.7/library/urllib.request.html#urllib-examples
-        encoded_url = urllib.parse.urlencode(data).encode('ascii')
-        req = urllib.request.Request(server, encoded_url, header)
```

Comment:
Updated the implementation from `urllib` to the `requests` library as it simplifies attaching files in multipart/form-data and provides built-in support for this format.